src/rpc/bootstrap_daemon.cpp

```diff
@@ -79,14 +93,17 @@ namespace cryptonote
 
   bool bootstrap_daemon::switch_server_if_needed()
   {
-    if (!m_get_next_public_node || m_http_client.is_connected())
+    if (m_http_client.is_connected() || !m_selector)
     {
       return true;
     }
 
-    const boost::optional<std::string> address = m_get_next_public_node();
-    if (address) {
-      return set_server(*address);
+    const auto node = [this]() {
+      const boost::unique_lock<boost::mutex> lock(m_selector_mutex);
+      return m_selector->next_node();
+    }();
+    if (node) {
+      return set_server(node->address, node->credentials);
```

Comment:
Is that a verbose version of:

const boost::unique_lock<boost::mutex> lock(m_selector_mutex);
const auto node = m_selector->next_node();
lock.unlock();
if (node) {
?

Reply:
Dropped the lambda